File: kirun-js/src/engine/function/system/array/Equals.ts
import { EventResult } from '../../../model/EventResult';
import { FunctionOutput } from '../../../model/FunctionOutput';
import { FunctionExecutionParameters } from '../../../runtime/FunctionExecutionParameters';
import { MapUtil } from '../../../util/MapUtil';
import { AbstractArrayFunction } from './AbstractArrayFunction';
import { Compare } from './Compare';

export class Equals extends AbstractArrayFunction {
    public constructor() {
        super(
            'Equals',
            [
                Equals.PARAMETER_ARRAY_SOURCE,
                Equals.PARAMETER_INT_SOURCE_FROM,
                Equals.PARAMETER_ARRAY_FIND,
                Equals.PARAMETER_INT_FIND_FROM,
                Equals.PARAMETER_INT_LENGTH,
            ],
            Equals.EVENT_RESULT_BOOLEAN,
        );
    }

    protected async internalExecute(context: FunctionExecutionParameters): Promise<FunctionOutput> {
        let compare: Compare = new Compare();

        let fo: FunctionOutput = await compare.execute(context);

        let resultMap: Map<string, any> = fo.allResults()[0].getResult();

        let v: number = resultMap.get(Equals.EVENT_RESULT_NAME);

        return new FunctionOutput([
            EventResult.outputOf(MapUtil.of(Equals.EVENT_RESULT_NAME, v == 0)),
        ]);
    }
}

File: kirun-java/src/main/java/com/fincity/nocode/kirun/engine/function/system/array/Equals.java
package com.fincity.nocode.kirun.engine.function.system.array;

import java.util.List;
import java.util.Map;

import com.fincity.nocode.kirun.engine.model.EventResult;
import com.fincity.nocode.kirun.engine.model.FunctionOutput;
import com.fincity.nocode.kirun.engine.runtime.reactive.ReactiveFunctionExecutionParameters;
import com.google.gson.JsonElement;
import com.google.gson.JsonPrimitive;

import reactor.core.publisher.Mono;

public class Equals extends AbstractArrayFunction {

	public Equals() {
		super("Equals", List.of(PARAMETER_ARRAY_SOURCE, PARAMETER_INT_SOURCE_FROM, PARAMETER_ARRAY_FIND,
		        PARAMETER_INT_FIND_FROM, PARAMETER_INT_LENGTH), EVENT_RESULT_BOOLEAN);
	}

	protected Mono<FunctionOutput> internalExecute(ReactiveFunctionExecutionParameters context) {

		Compare compare = new Compare();

		return compare.execute(context)
		        .map(fo ->
				{
			        Map<String, JsonElement> resultMap = fo.allResults()
			                .get(0)
			                .getResult();

			        int v = resultMap.get(EVENT_RESULT_NAME)
			                .getAsJsonPrimitive()
			                .getAsInt();

			        return new FunctionOutput(
			                List.of(EventResult.outputOf(Map.of(EVENT_RESULT_NAME, new JsonPrimitive(v == 0)))));
		        });
	}

}
